src/components/Tabs/Annotation/AnnotationDisplay/AnnotationDisplayList.tsx

```diff
@@ -104,23 +142,43 @@ export default function AnnotationDisplayList(props: AnnotationDisplayListProps)
           }}
         >
           <FlexRowAlignCenter style={{ marginBottom: "5px" }} $gap={10}>
-            <AnnotationTrackThumbnail
-              frame={props.frame}
-              setFrame={props.setFrame}
-              ids={selectedTrackIds}
-              track={props.selectedTrack}
-              dataset={props.dataset}
-              color={props.labelColor}
-              mark={markedTime}
-              highlightedIds={highlightRange}
-            ></AnnotationTrackThumbnail>
+            <TooltipContainer ref={tooltipContainerRef} $x={thumbnailHoveredX ?? lastHoveredX.current}>
+              <AnnotationTrackThumbnail
+                frame={props.frame}
+                setFrame={props.setFrame}
+                onHover={(x, time) => {
+                  setThumbnailHoveredX(x);
+                  setThumbnailHoveredTime(time);
+                  if (x !== null) {
+                    lastHoveredX.current = x;
+                  }
+                }}
+                ids={highlightedIds}
+                bgIds={bgIds}
+                track={props.selectedTrack}
+                dataset={props.dataset}
+                color={props.labelColor}
+                mark={markedTime}
+                highlightedIds={highlightRange}
+              ></AnnotationTrackThumbnail>
```

Comment:
just a bit confused here, we're passing `highlightedIds` to `ids` but `highlightRange` to `highlightedIds`? Might be worth renaming one?

Reply:
Oops, good point! `highlightedIds` will be renamed.